src/external/Eigen/CMakeLists.txt

```diff
@@ -0,0 +1,3 @@
+project(TuriExternalDependencies)
+
+make_empty_library(eigen)
```

Comment:
Why is it empty? Is Eigen a header-only library? If so, you may not even need a target for it at all.

Reply:
True.  Earlier versions had a small configuration step; that is gone now.  IMO it's better to declare the dependency explicitly if possible.  That makes it easier to, e.g. move the package to the deps/ folder.

Reply:
🤷‍♂️ 

Personally I don't see why we would want to move this to the deps folder, or why we want anything to explicitly depend on an empty library, but it doesn't really hurt either.

Note that in #1403 I'm actually removing most of the existing empty libraries, as that change splits compile and link-time dependencies apart, and the empty libraries are neither a compile-time nor a link-time dependency of... well, anything, really.

Reply:
I see what you're saying.  I'm okay doing it that way.